### PR TITLE
Updated Twilio Client for Android Quickstart to work with new Server Apps

### PR DESCRIPTION
**Pull request by jefflinwood**
This PR lets the Twilio Client for Android work with the new Twilio Client server quickstarts - biggest change is that the identity is generated along with the token, so the client doesn't create its own identity any more.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,55 @@
...
+* [Instructions for Node.js](https://github.com/TwilioDevEd/client-quickstart-node)
+* [Instructions for PHP](https://github.com/TwilioDevEd/client-quickstart-php)
+* [Instructions for Python](https://github.com/TwilioDevEd/client-quickstart-python)
+* [Instructions for Ruby](https://github.com/TwilioDevEd/client-quickstart-ruby)
+
+Your application should now be running locally. 
+
+## Host Your Server
+Once you've got your server up and running, you will need to host it somewhere Twilio can access it. You can publish your app to a web hosting provider, or you can run it locally and use [ngrok](https://ngrok.com/) to create a tunnel to your development machine with a publicly accessible URL. See the README above for the specific ngrok instructions for your preferred platform.
+
```
Comment (vassjozsef) [inline README.md]: I am -1 on this. Please see https://github.com/twilio/mobile-quickstart, where you can host on heroku with a one-click install.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+Choose a download package for your server-side language of choice. If you're primarily a mobile app developer and don't have a strong preference, Node.js or Ruby will probably get you up and running the fastest.
+
+- [Download for C#](https://github.com/TwilioDevEd/client-quickstart-csharp/archive/master.zip)
+- [Download for Java](https://github.com/TwilioDevEd/client-quickstart-java/archive/master.zip)
+- [Download for Node.js](https://github.com/TwilioDevEd/client-quickstart-node/archive/master.zip)
+- [Download for PHP](https://github.com/TwilioDevEd/client-quickstart-php/archive/master.zip)
+- [Download for Python](https://github.com/TwilioDevEd/client-quickstart-python/archive/master.zip)
+- [Download for Ruby](https://github.com/TwilioDevEd/client-quickstart-ruby/archive/master.zip)
+
```
Comment (vassjozsef) [inline README.md]: As a mobile developer, it is probably not that important to me that the server-side code is available in several languages. What is important that I can setup the server side as easy as possible and forget about it.

File: README.md (hunk 1 above)
Comment (kwhinnery) [inline README.md]: We have behavior data from the docs suggesting that the Heroku button does not get used, so we don't as a general rule provide them on sample apps anymore. The ngrok route serves to give the developer a productive local environment setup that they can modify the server app without a lengthy Heroku deploy.

**Comment by vassjozsef** [comment]
I am not a mobile developer so please take my comments accordingly. The Heroku button idea originated from Twilio.org, where even today projects has this options. https://github.com/twilio/mobile-quickstart offers both running it locally and using `ngrok` and deploying to Heroku.

We wanted to make it easier to mobile developers to start with Twilio, as this being a common complaint. We majorly revised Quickstart document, we included two sample apps (Quickstart and BasicPhone) and provided a single Python server that worked with both Andoid/iOS sample apps that you can run locally or host with a single click.

I do not think that a Swift developer want to tweak the ruby script when just starting to experiment with SDK. For me, it took nearly half a day to make my first call since I did not understand why I need a web-server to make a phone call.

Maybe providing ready to go sample apps within the console is the way to go.

**Comment by kbagchiGWC** [comment]
+1 on a sample app idea on the console.

**Comment by fabriziomoscon** [comment]
I might join the party a bit too late, but I also think there could be a faster way for a Native developer to get a Twilio server up and running.
In an ideal world I would like to have a docker image containing a Twilio server running code. The python one would be perfect, but nothing stop creating docker images for any possible Twilio backend. As a plus this image would work on dev, stage and prod environments.
The image could get started with:
`docker run -d -p "80:5000" -name=twilio-server twilio/server-python --env-file=".env"`
where `.env` contains:
```
ACCOUNT_SID=
AUTH_TOKEN=
APP_SID=
CALLER_ID=
```
Also `ngrok` could be installed and run in the container. So the developer could find the public url inside the docker logs.

After running the demo the developer would simply stop the container and nothing would be left on the host machine, since the server was conteinerized.

I tried before the Heroku button, but I am not a big fun of it. From my experience if you are new to Heroku you don't quite know what happens after the click. At least Docker can be installed locally and it is getting more and more popular every day. In top of that I think that iterations on a docker build are faster that on a remote Heroku (or another vendor) deploy.

Even for those developers new to docker. It will take less than half a day to have a server running.

**Comment by idelgado** [comment]
Closing this PR. We will be recommending developers use `Voice Android` moving forward. https://github.com/twilio/voice-quickstart-android

**Pull request closed without merging**